src/main/scala/diplomacy/Nodes.scala

```diff
@@ -90,38 +454,66 @@ abstract class BaseNode(implicit val valName: ValName)
     * by taking the name of the node as determined from valName,
     * converting camel case into snake case, and stripping "Node" or "NodeOpt" suffixes.
     */
-  def wirePrefix = {
+  def wirePrefix: String = {
     val camelCase = "([a-z])([A-Z])".r
     val decamel = camelCase.replaceAllIn(valName.name, _ match { case camelCase(l, h) => l + "_" + h })
     val name = decamel.toLowerCase.stripSuffix("_opt").stripSuffix("node").stripSuffix("_")
     if (name.isEmpty) "" else name + "_"
   }
 
+  /** @return Node description, which should defined by subclasses. */
   def description: String
   def formatNode: String = ""
 
+  /** @return metedata to visualize inward edges into this node. */
   def inputs:  Seq[(BaseNode, RenderedEdge)]
+  /** @return metedata to print outward edges from this node. */
   def outputs: Seq[(BaseNode, RenderedEdge)]
 
+  /** @todo
+    * [[MixedAdapterNode]] will be true, otherwise will be false
+    */
   protected[diplomacy] def flexibleArityDirection: Boolean = false
```

Comment:
I still don't understand what arity direction means?
I can only see it is overwritten in `MixedAdapterNode`

Reply:
["Arity"](https://en.wikipedia.org/wiki/Arity) is a word that essentially means "number of" in a few different areas of mathematics and computer science.

When talking about functions, it means the number of arguments to the function. When talking about graphs, it's usually about the number if edges coming out (or going in) to a node. I'm not specifically familiar with this code, so I don't know the specific meaning, but you should be able to figure it out from reading where it is used.